### PR TITLE
Link improvements, mainly for customers not logged in

### LINKED ISSUE #24999: Improve my account links in footer
### Is your feature request related to a problem?

Links are the same for logged in customer and a guest. The guest should could have a nice link to track his order, log in or register instead.

### Before

current

### After, logged in

after_logged

### After, logged out

after_notlogged

### PRs
Module - https://github.com/PrestaShop/ps_customeraccountlinks/pull/34
Theme - https://github.com/PrestaShop/PrestaShop/pull/25064

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | I provided some useful links to customers that either do not have an account or are not logged in. Related Classic theme PR here: https://github.com/PrestaShop/PrestaShop/pull/25064
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#24999.
| How to test?  | See the links when not logged in and when logged in. Also, try to install wishlist module, so you can see that hooked modules show only for logged in customers.

logged
notlogged

### CODE REVIEW

File: ps_customeraccountlinks.tpl
Comment (PierreRambaud) [inline ps_customeraccountlinks.tpl]:
```suggestion
  <ul>
    {if $logged}
      {foreach from=$my_account_urls item=my_account_url}
        <li><a href="{$my_account_url.url}" title="{$my_account_url.title}" rel="nofollow">{$my_account_url.title}</a></li>
      {/foreach}
      {hook h="displayMyAccountBlock"}
      <li><a href="{$logout_url}" title="{l s='Log me out' d='Shop.Theme.Customeraccount'}" rel="nofollow">{l s='Sign out' d='Shop.Theme.Actions'}</a></li>
    {else}
      <li><a href="{$urls.pages.guest_tracking}" title="{l s='Guest Tracking' d='Shop.Theme.Customeraccount'}" rel="nofollow">{l s='Guest Tracking' d='Shop.Theme.Customeraccount'}</a></li>
      <li><a href="{$urls.pages.my_account}" title="{l s='Log in to your customer account' d='Shop.Theme.Customeraccount'}" rel="nofollow">{l s='Sign in' d='Shop.Theme.Actions'}</a></li>
      <li><a href="{$urls.pages.register}" title="{l s='Create an account' d='Shop.Theme.Customeraccount'}" rel="nofollow">{l s='Create an account' d='Shop.Theme.Customeraccount'}</a></li>
    {/if}
  </ul>
```

**Review by PierreRambaud — DISMISSED** [review]

File: ps_customeraccountlinks.tpl
Comment (PierreRambaud) [inline ps_customeraccountlinks.tpl:32]: You remove the hook?

Comment (Hlavtox) [reply ps_customeraccountlinks.tpl:32]: @PierreRambaud Its there, just moved

**Comment by khouloudbelguith** [comment]
Hi @Hlavtox,

This PR should be reviewed by our devs First.
Ping  @PrestaShop/prestashop-maintainers 

Thank you!

**Comment by marionf** [comment]
Hello @Hlavtox 

The "guest tracking" link will be displayed just for the guests who ordered something, right?

**Comment by Hlavtox** [comment]
@PierreRambaud No BC breaks, all the variables used in the template were already present in the minimal supported version of Prestashop.

File: ps_customeraccountlinks.tpl
Comment (PierreRambaud) [inline ps_customeraccountlinks.tpl:27]: Using my_account_urls allows people to add more value from hooks. Not sure putting all links here is the right solution

Comment (Hlavtox) [reply ps_customeraccountlinks.tpl:27]: @PierreRambaud They should use {hook h="displayMyAccountBlock"} for that, and not overriding the module. (how would that be possible? :-))

Comment (PierreRambaud) [reply ps_customeraccountlinks.tpl:27]: Everything is possible with PrestaShop you know ;)

**Comment by Julievrz** [comment]
Please see [my comment](https://github.com/PrestaShop/PrestaShop/pull/25064#issuecomment-875469571) about wording. :pray:

**Comment by Hlavtox** [comment]
@Julievrz @marionf Solved

File: ps_customeraccountlinks.tpl
Comment (Julievrz) [inline ps_customeraccountlinks.tpl]:
```suggestion
      <li><a href="{$urls.pages.guest_tracking}" title="{l s='Order tracking' d='Shop.Theme.Customeraccount'}" rel="nofollow">{l s='Order tracking' d='Shop.Theme.Customeraccount'}</a></li>
```

**Comment by Julievrz** [comment]
Wording ✅ Thank you!

File: ps_customeraccountlinks.tpl
Comment (Hlavtox) [reply ps_customeraccountlinks.tpl:27]: @PierreRambaud We have all the URLs generated from the core, I don't see why we should keep some legacy code and slow down the development because of wrong practices. 😛

They have a nice hook that every module uses. ;-)

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@PierreRambaud The tests still don't work, should I rebase it? 🤔

**Comment by PierreRambaud** [comment]
> @PierreRambaud The tests still don't work, should I rebase it? thinking

Think, it's a good idea

**Comment by florine2623** [comment]
Hello @Hlavtox ,

I hope you are doing OK !
The checks are not working, have you checked for the rebase ?

Thanks!

**Comment by Hlavtox** [comment]
Hi! I merged the commits together. Try to approve and let's see if it works. The branch is equal to dev.

**Review by Progi1984 — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @Progi1984 ,

I see that some checks were not successful ?
Is this an issue with CI ?
Thanks!

**Comment by Progi1984** [comment]
@Hlavtox Could you fix PHStan ?

Thanks @florine2623

**Comment by Hlavtox** [comment]
@Progi1984 Dont know how, can you help? 🙏

**Comment by PierreRambaud** [comment]
It seems that the PHPStan error is not related to this PR.
If it happens again, please rebase your pull request :)

**Comment by PierreRambaud** [comment]
The CI is broken and had to be fixed :sweat_smile:

**Comment by Hlavtox** [comment]
@PierreRambaud The PR is up to date man :( `This branch is 2 commits ahead of PrestaShop:dev.`

**Comment by PierreRambaud** [comment]
> @PierreRambaud The PR is up to date man :( `This branch is 2 commits ahead of PrestaShop:dev.`

Yes but CI is failing, it has to be fixed before

**Comment by florine2623** [comment]
Hello @Hlavtox and all,

Thanks for the update on this !
I've tested this issue in combination with https://github.com/PrestaShop/PrestaShop/pull/25064, it looks good to me !
It is QA approved !

**Comment by PierreRambaud** [comment]
#35 will fix the CI

**Comment by PierreRambaud** [comment]
Thank you @Hlavtox

**Pull request merged**
